src/debugger/core.clj

```diff
@@ -55,20 +86,34 @@
                  macro-ns# (ns-name (or (:ns outer-fn-meta#) *ns*))
                  macro-eval-fn# (partial eval-fn macro-ns# macro-break-line# outer-fn-symbol# trace# signal-val# return-val# cached-cont-val# locals-fn# cont-fn#)
                  macro-read-fn# (partial read-fn signal-val#)
-                 macro-prompt-fn# (partial prompt-fn outer-fn-symbol# macro-break-line# signal-val#)]
-             (println "\nBreak from:" outer-fn-path# "(type \"(help)\" for help)")
-             (print-short-source macro-break-line# outer-fn-symbol#)
-             (clojure.main/repl
-               :prompt macro-prompt-fn#
-               :eval macro-eval-fn#
-               :read macro-read-fn#
-               :caught caught-fn)
-             (or
-               (deref return-val#)
-               (deref cached-cont-val#)
-               (cont-fn#))))
-         ;; not repl or skip
-         (do ~@body)))))
+                 macro-prompt-fn# (partial prompt-fn outer-fn-symbol# macro-break-line# signal-val#)
+                 return# (promise)]
+             (letfn [(breakpoint# []
```

Comment:
I use `letfn` here because I need to reference `breakpoint#` from inside `breakpoint#` in order to deregister it.

Which would you prefer?

1) I try to refactor so `breakpoint#` is never called from `breakpoint#` and move it into the let.
2) Move `macro-eval-fn#` and `macro-read-fn#` into the letfn

Also in writing this, I realize I should rename `breakpoint#` to `breakpoint-fn#` :)